iceoryx2/src/service/attribute.rs

```diff
@@ -98,26 +98,60 @@
 //! ```
 
 use core::ops::Deref;
+
+use iceoryx2_bb_container::{byte_string::FixedSizeByteString, vec::FixedSizeVec};
+use iceoryx2_bb_derive_macros::ZeroCopySend;
 use iceoryx2_bb_elementary::CallbackProgression;
+use iceoryx2_bb_log::fatal_panic;
 use serde::{Deserialize, Serialize};
 
+use crate::constants::{MAX_ATTRIBUTES, MAX_ATTRIBUTE_KEY_LENGTH, MAX_ATTRIBUTE_VALUE_LENGTH};
+
+type AttributeKeyString = FixedSizeByteString<MAX_ATTRIBUTE_KEY_LENGTH>;
+type AttributeValueString = FixedSizeByteString<MAX_ATTRIBUTE_VALUE_LENGTH>;
+
+type KeyStorage = FixedSizeVec<AttributeKeyString, MAX_ATTRIBUTES>;
+type AttributeStorage = FixedSizeVec<Attribute, MAX_ATTRIBUTES>;
+
 /// Represents a single service attribute (key-value) pair that can be defined when the service
 /// is being created.
-#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Clone, PartialOrd, Ord)]
+#[derive(Debug, Eq, PartialEq, Clone, PartialOrd, Ord, ZeroCopySend, Serialize, Deserialize)]
+#[repr(C)]
 pub struct Attribute {
-    key: String,
-    value: String,
+    key: AttributeKeyString,
+    value: AttributeValueString,
+}
+
+/// Errors that can occur when creating an [`Attribute`].
+pub enum AttributeError {
+    /// The attribute key exceeds the maximum allowed length.
+    KeyTooLong,
+    /// The attribute value exceeds the maximum allowed length.
+    ValueTooLong,
 }
 
 impl Attribute {
+    /// Creates an attribute instance
+    pub fn new(key: &str, value: &str) -> Result<Self, AttributeError> {
```

Comment:
This here is fine as it is since it returns an `AttributeError` when the key or value is too long.

Reply:
To be consistent with the rest of the methods, I made the arguments the fixed strings and changed the return type to `Self`.

The capacity violations will now be conveyed by the `FixedSizeByteString` arguments.